src/player.js

```diff
@@ -1811,13 +1811,15 @@ export default class Player extends FakeEventTarget {
         }
       });
       if (this.config.playback.playAdsWithMSE) {
-        this._eventManager.listen(this, AdEventType.AD_LOADED, (event: FakeEvent) => {
+        this._eventManager.listen(this, AdEventType.AD_LOADED, event => {
           if (event.payload.ad.linear) {
-            this._detachMediaSource();
+            this._eventManager.listenOnce(this, AdEventType.AD_BREAK_START, () => {
+              this._detachMediaSource();
+            });
```

Comment:
do we still need the AD_LOADED? why not register to AD_BREAK_START only?
If it is because of the ad.linear that only comes in AD_LOADED then I assume we can take the ad type from the current ad we have in the ad controller.